### PR TITLE
feature/separate-pfdhcplistener-queues

### LINKED ISSUE #2859: pfqueue: Separate pfdhcplistener queues for internal and external traffic
Right now the traffic coming on the management interface (and dhcplistener) use the same queue as the internal networks of PacketFence.

Given we might want to assign different weights or different amount of dedicated workers to both types of traffic, it would be nice to split those two different type of traffic into two different queues

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
===========
Separate pfdhcplistener queues for internal and external traffic

Impacts
=======
pfqueue and pfdhcplistener

Issue
=====
Fixes #2859

Delete branch after merge
=========================
NO

NEWS file entries
=================

Enhancements
------------
* Separate pfdhcplistener queues for internal and external traffic

### CODE REVIEW

File: sbin/pfdhcplistener [hunk 1]
```diff
@@ -175,7 +177,21 @@ sub setup_global {
...
+
+=head2 which_send_queue
+
+Determine which send queue
 
+=cut
+
+sub which_send_queue {
+    my ($net_type) = @_;
+    return (any { exists $NET_TYPES_FOR_EXTERNAL_QUEUE{$_} } expand_csv($net_type) ) ? 'pfdhcplistener_external' : 'pfdhcplistener' ;
```
Comment (julsemaan) [inline sbin/pfdhcplistener:193]: From my testing %NET_TYPES_FOR_EXTERNAL_QUEUE is empty here which prevents the detection from working

Comment (jrouzierinverse) [reply sbin/pfdhcplistener:193]: Fixed by 4dfa485

File: lib/pf/util.pm [hunk 2]
```diff
@@ -940,6 +941,24 @@ sub trim_path {
    return ((@parts == 0) ? '' : catdir(@parts));
 }
 
+
+=item expand_csv
+
```
Comment (julsemaan) [inline lib/pf/util.pm:946]: Some little perldoc would be good

File: sbin/pfdhcplistener [hunk 3]
```diff
@@ -175,7 +182,16 @@ sub setup_global {
     return ;
 }
 
+=head2 which_send_queue
+
+Determine which send queue
```
Comment (julsemaan) [inline sbin/pfdhcplistener:187]: Rephrase that, doesn't make much sense

**Comment by extrafu** [comment]
@julsemaan Same thing as PR #2887

**Comment by julsemaan** [comment]
@extrafu 

Just need the adjustments made and we're good for merge

**Comment by extrafu** [comment]
@jrouzierinverse Do the last changes and inform us when done. We need to merge this.

**Comment by jrouzierinverse** [comment]
Will rework when #2887 is merged

**Comment by jrouzierinverse** [comment]
Sorry this is included in #2887

**Pull request closed without merging**